### DISCUSSION TITLE
How can I select the label of the Zilla pod? Making Zilla accessible outside of the Kubernetes (k8s) cluster using a service.

### DISCUSSION THREAD

**Discussion by MartinMohammed** [aklivity/zilla#1125] [Q&A] [answered]
Suppose I want to use a Kubernetes (k8s) service to expose Zilla to the public network (outside of the k8s cluster). How can I select the label of the Zilla pod?

A quick and dirty solution I found involves using k9s, a tool to interact with the k8s cluster. I can watch the deployment YAML file for the Zilla application and then select those labels with the selector in the service:

![image](https://github.com/aklivity/zilla/assets/81469658/661a0c37-c345-4327-a2e7-6d4031bd890e)

However, is there a better solution?

**Comment by vordimous** [comment, marked as answer]
The `app.kubernetes.io/*` labels are added by the helm install and can be updated for each deployment. 

```
app.kubernetes.io/name: {{ include "zilla.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
```

The "zilla.name" comes from the Chart and can be overritten in the values.yaml `nameOverride` property.

The ".Release.Name" property is the name you used when deploying the zilla helm chart. to change that value you can install zilla and use a different deployment name `helm install my-custom-name-for-zilla oci://ghcr.io/aklivity/charts/zilla`.

They are currently the same because the default zilla name and release name are the same.

**Reply by vordimous** [reply]
Additionally you can specifcy custom labels and annotations for the zilla pods in the `values.yaml`:

```
podAnnotations: {}
podLabels: {}
```
